File: lab_2/Sequence.cpp
#include <iostream>
#include <fstream>
#include <cstdlib>
#include <ctime>

/**
 * Generates a random binary sequence of length 128 bits
 * and saves it to the file "cpp_seq.txt".
 *
 * This program initializes the random number generator using the current time
 * and generates a sequence of 0 or 1.
 * The generated sequence is written to a text file named "cpp_seq.txt".
 *
 * @return int Returns 0 upon successful completion of the program.
 */
int main() {
    std::srand(static_cast<unsigned int>(std::time(0)));
    std::ofstream outFile("cpp_seq.txt");
    for (int i = 0; i < 128; i++)
        outFile << std::rand() % 2;
    outFile.close();
    return 0;
}

File: lab_2/Sequence.java
import java.io.BufferedWriter;
import java.io.FileWriter;
import java.util.Random;

/**
 * Generates a random binary sequence of length 128 bits
 * and saves it to the file "random_sequence.txt".
 *
 * This program uses the Random class to generate random sequence of 0 or 1.
 *
 * @throws Exception if an I/O error occurs while writing to the file
 */
public class Sequence {
    public static void main(String[] args) throws Exception {
        StringBuilder binarySequence = new StringBuilder(128);
        Random random = new Random();
        for (int i = 0; i < 128; i++)
            binarySequence.append(random.nextInt(2));
        BufferedWriter writer = new BufferedWriter(new FileWriter("random_sequence.txt"));
        writer.write(binarySequence.toString());
        writer.close();
    }
}
